elf2table/src/lib.rs

```diff
@@ -1,17 +1,15 @@
 //! Reads ELF metadata and builds an interner table
 
-use std::collections::BTreeMap;
+use std::{borrow::Cow, collections::BTreeMap, path::PathBuf};
 
-use anyhow::anyhow;
+use anyhow::{anyhow, bail, ensure};
 pub use decoder::Table;
-use object::{File, Object, ObjectSection};
+use object::{Object, ObjectSection};
 
 /// Parses an ELF file and returns the decoded `defmt` table
 ///
 /// This function returns `None` if the ELF file contains no `.defmt` section
-pub fn parse(elf: &[u8]) -> Result<Option<Table>, anyhow::Error> {
-    let elf = File::parse(elf)?;
-
+pub fn parse(elf: &object::File) -> Result<Option<Table>, anyhow::Error> {
```

Comment:
now I'm no long sure if changing `[u8]` to `object::File` here was the right call. It avoids re-parsing the ELF in `probe-run` but makes this API (semver) depend on the version of `object` as one of its types is publicly exposed. This would be the only type from a dependency that's exposed in the public API (I think)

Reply:
Yeah I went from `xmas_elf::ElfFile` to `&[u8]` when switching to `object` to avoid having to care about public dependencies.

I expect host-side tools to always be built optimized, so parsing the ELF headers twice should have negligible impact.